src/foam/core/pool/ThreadPoolAgency.js

```diff
@@ -109,37 +109,32 @@ foam.CLASS({
       javaCode: `
     Loggers.logger(getX(), this).info(getPrefix(), "start");
     threadGroup_ = new ThreadGroup(Thread.currentThread().getThreadGroup(), getPrefix());
-    pool_ = new ThreadPoolExecutor(
-      getNumberOfThreads(),
-      getNumberOfThreads(),
-      10,
-      TimeUnit.SECONDS,
-      new LinkedBlockingQueue<Runnable>(),
-      new ThreadFactory() {
-        final AtomicInteger threadNumber = new AtomicInteger(1);
-
-        public Thread newThread(Runnable runnable) {
-          Thread thread = new Thread(
-            threadGroup_,
-            runnable,
-            getPrefix() + "-" + threadNumber.getAndIncrement(),
-            0
-            );
-          // Thread does not block server from shut down.
-          thread.setDaemon(true);
-          thread.setPriority(Thread.NORM_PRIORITY);
-          return thread;
-        }
-      }
-    );
-    pool_.allowCoreThreadTimeOut(true);
+    initThreadPool();
     scheduleReporting();
     `
     },
+    {
+      name: 'stop',
+      javaCode: `
+        getPool().shutdownNow();
+        synchronized ( queuedLock_ ) {
+          clearQueued();
+        }
+        synchronized ( executedLock_ ) {
+          clearExecuted();
+        }
+        synchronized ( executingLock_ ) {
+          clearExecuting();
+        }
+      `
+    },
     {
       name: 'reload',
       javaCode: `
+      stop();
       Loggers.logger(getX(), this).info(getPrefix(), "reload");
+      threadGroup_ = new ThreadGroup(Thread.currentThread().getThreadGroup(), getPrefix());
```

Comment:
just a suggestion, I think we can put it in initThreadPoll method as well

Reply:
Fixed.